features/retry.feature

```diff
@@ -0,0 +1,444 @@
+  Scenario: running Cucumber JS with --retryTagFilter but no positive --retry will fail
+    When I run cucumber-js with `--retryTagFilter @flaky`
+    Then the error output contains the text:
+      """
+      Error: a positive --retry count must be specified when setting --retryTagFilter
+      """
+    And it fails
+
+  Scenario: running Cucumber JS with negative --retry will fail
+    When I run cucumber-js with `--retry -1`
+    Then the error output contains the text:
+      """
+      Error: --retry must be a non negative integer
+      """
+    And it fails
+
+  Scenario: running Cucumber JS with --retry 0 will let a failing test fail and not retry
+    Given a file named "features/a.feature" with:
+      """
+      Feature:
+        Scenario: Failing
+          Given a failing step
+      """
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {Given} from 'cucumber'
+
+      Given(/^a failing step$/, function() { throw 'fail' })
+      """
+    When I run cucumber-js with `--retry 0`
+    Then the output contains the text:
+      """
+      F
+
+      Failures:
+
+      1) Scenario: Failing # features/a.feature:2
+      ✖ Given a failing step # features/step_definitions/cucumber_steps.js:3
+      Error: fail
+      """
+    And scenario "Failing" step "Given a failing step" failed with:
+      """
+      Error: fail
+      """
+    And it fails
+
+  Scenario: running Cucumber JS without --retry will let a failing test fail and not retry
+    Given a file named "features/a.feature" with:
+      """
+      Feature:
+        Scenario: Failing
+          Given a failing step
+      """
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {Given} from 'cucumber'
+
+      Given(/^a failing step$/, function() { throw 'fail' })
+      """
+    When I run cucumber-js
+    Then the output contains the text:
+      """
+      F
+
+      Failures:
+
+      1) Scenario: Failing # features/a.feature:2
+      ✖ Given a failing step # features/step_definitions/cucumber_steps.js:3
+      Error: fail
+      """
+    And scenario "Failing" step "Given a failing step" failed with:
+      """
+      Error: fail
+      """
+    And it fails
+
+  Scenario: retrying a flaky test will eventually make it pass
+    Given a file named "features/a.feature" with:
+      """
+      Feature:
+        Scenario: Flaky
+          Given a flaky step
+      """
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {Given} from 'cucumber'
+
+      let willPass = false
+
+      Given(/^a flaky step$/, function() {
+        if (willPass) {
+          return
+        }
+        willPass = true
+        throw 'fail'        
+      })
+      """
+    When I run cucumber-js with `--retry 1`
+    Then it outputs the text:
+      """
+      F.
+      
+      Warnings:
+      
+      1) Scenario: Flaky (attempt 2) # features/a.feature:2
```

Comment:
I think it would be more useful to output any failed attempt that are retried as warnings. The line could read something like `1) <scenario name> (attempt X, retried)`. I don't think outputting the successful attempt is as useful.

Reply:
For awareness I started working on this and will try to push up my updates today